test/org.eclipse.elk.graph.json.test/src/org/eclipse/elk/graph/json/test/EdgeCoordsTest.xtend

```diff
@@ -0,0 +1,176 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.elk.graph.json.test
+
+import com.google.gson.JsonParser
+import org.eclipse.elk.core.RecursiveGraphLayoutEngine
+import org.eclipse.elk.core.util.BasicProgressMonitor
+import org.eclipse.elk.core.util.Maybe
+import org.eclipse.elk.graph.json.ElkGraphJson
+import org.eclipse.elk.graph.json.JsonImporter
+import org.junit.Test
+
+import static org.junit.Assert.*
+
+/**
+ */
+class EdgeCoordsTest {
+    
+    val graph = '''
+    {
+        "id": "root",
+        "properties": {
+            "algorithm": "layered",
+            "org.eclipse.elk.hierarchyHandling": "INCLUDE_CHILDREN"
+        },
+        "children": [
+            { "id": "A",
+              "children": [
+                  { "id": "x", "width": 50, "height": 90 },
+                  { "id": "B",
+                    "children": [
+                        { "id": "y", "width": 50, "height": 90 },
+                        { "id": "z", "width": 50, "height": 90 }
+                    ],
+                    "edges": [
+                        { "id": "e1", "sources": [ "y" ], "targets": [ "z" ] },
+                        { "id": "e2", "sources": [ "x" ], "targets": [ "z" ] }
+                    ]
+                  }
+              ]
+            }
+        ]
+    }
+    '''
+    
+    val secContainer = '''
+    {
+     "id": "e2_s0",
+     "startPoint": {
+      "x": 62,
+      "y": 124
+     },
+     "endPoint": {
+      "x": 169,
+      "y": 99
+     },
+     "bendPoints": [
+      {
+       "x": 159,
+       "y": 124
+      },
+      {
+       "x": 159,
+       "y": 99
+      }
+     ],
+     "incomingShape": "x",
+     "outgoingShape": "z"
+    }
+    '''
+    
+    val secParent = '''
+    {
+     "id": "e2_s0",
+     "startPoint": {
+      "x": -25,
+      "y": 112
+     },
+     "endPoint": {
+      "x": 82,
+      "y": 87
+     },
+     "bendPoints": [
+      {
+       "x": 72,
+       "y": 112
+      },
+      {
+       "x": 72,
+       "y": 87
+      }
+     ],
+     "incomingShape": "x",
+     "outgoingShape": "z"
+    }
+    '''
+    
+    val secRoot = '''
+    {
+     "id": "e2_s0",
+     "startPoint": {
+      "x": 74,
+      "y": 136
+     },
+     "endPoint": {
+      "x": 181,
+      "y": 111
+     },
+     "bendPoints": [
+      {
+       "x": 171,
+       "y": 136
+      },
+      {
+       "x": 171,
+       "y": 111
+      }
+     ],
+     "incomingShape": "x",
+     "outgoingShape": "z"
+    }
+    '''
+            
+    @Test
+    def void edgeCoordsTest() {
+        
+        val cases = #[
+            #['CONTAINER', secContainer],
+            #['PARENT', secParent],
+            #['ROOT', secRoot]
+        ]
+        
+        for (p : cases) {
+            val mode = p.get(0)
+            val expectedString = p.get(1)
+            
+            val parser = new JsonParser()
```

Comment:
I suggest using the static method `JsonParser.parseString(...)` instead of the deprecated constructor and `parse` method.